### PR TITLE
World generation mappings

### PR DESCRIPTION
**Pull request by mschae23**
I did some more world generation mappings.

### CODE REVIEW

File: mappings/net/minecraft/world/gen/GeneratorOptions.mapping [hunk 1]
```diff
@@ -38,9 +38,17 @@ CLASS net/minecraft/class_5285 net/minecraft/world/gen/GeneratorOptions
...
 		ARG 2 seed
+	METHOD method_28608 getRegistryWithReplacedOverworldGenerator (Lnet/minecraft/class_2378;Lnet/minecraft/class_2370;Lnet/minecraft/class_2794;)Lnet/minecraft/class_2370;
+		ARG 0 dimensionTypeRegistry
+		ARG 1 optionsRegistry
+		ARG 2 generator
 	METHOD method_28609 getDimensions ()Lnet/minecraft/class_2370;
 	METHOD method_28610 validate ()Lcom/mojang/serialization/DataResult;
 	METHOD method_28611 isStable ()Z
 	METHOD method_29575 getWorlds ()Lcom/google/common/collect/ImmutableSet;
+	METHOD method_29962 getRegistryWithReplacedOverworld (Lnet/minecraft/class_2370;Ljava/util/function/Supplier;Lnet/minecraft/class_2794;)Lnet/minecraft/class_2370;
```
Comment (mschae23) [inline mappings/net/minecraft/world/gen/GeneratorOptions.mapping:49]: I'm not so sure about this name, it's very long. Should I change it to something like `withReplacedOverworld`, or should I keep it like this?

Comment (liach) [reply mappings/net/minecraft/world/gen/GeneratorOptions.mapping:49]: I think this name is good enough. Someone else is free to suggest more concise names.

File: mappings/net/minecraft/world/gen/GeneratorOptions.mapping [hunk 2]
```diff
@@ -38,9 +38,17 @@ CLASS net/minecraft/class_5285 net/minecraft/world/gen/GeneratorOptions
...
+		ARG 0 optionsRegistry
+		ARG 1 dimensionType
+		ARG 2 generator
```
Comment (liach) [inline mappings/net/minecraft/world/gen/GeneratorOptions.mapping]:
```suggestion
		ARG 1 overworldDimensionType
		ARG 2 overworldGenerator
```
To be more informative.

File: mappings/net/minecraft/world/gen/GeneratorOptions.mapping [hunk 3]
```diff
@@ -38,9 +38,17 @@ CLASS net/minecraft/class_5285 net/minecraft/world/gen/GeneratorOptions
 		ARG 0 biomeRegistry
 		ARG 1 chunkGeneratorSettingsRegistry
 		ARG 2 seed
+	METHOD method_28608 getRegistryWithReplacedOverworldGenerator (Lnet/minecraft/class_2378;Lnet/minecraft/class_2370;Lnet/minecraft/class_2794;)Lnet/minecraft/class_2370;
+		ARG 0 dimensionTypeRegistry
+		ARG 1 optionsRegistry
+		ARG 2 generator
```
Comment (liach) [inline mappings/net/minecraft/world/gen/GeneratorOptions.mapping]:
```suggestion
		ARG 2 overworldGenerator
```

**Review by jaskarth — APPROVED** [review]

**Comment by liach** [comment]
This needs an update to fix the merge conflict.

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/structure/StructureStart.mapping [hunk 4]
```diff
@@ -42,6 +48,11 @@ CLASS net/minecraft/class_3449 net/minecraft/structure/StructureStart
 		ARG 1 registryManager
 		ARG 2 chunkGenerator
 		ARG 3 manager
+		ARG 4 pos
+		ARG 5 biome
+		ARG 6 config
+		ARG 7 heightLimitView
```
Comment (YanisBft) [inline mappings/net/minecraft/structure/StructureStart.mapping]:
```suggestion
		ARG 7 world
```

File: mappings/net/minecraft/world/gen/chunk/FlatChunkGeneratorConfig.mapping [hunk 5]
```diff
@@ -34,3 +34,10 @@ CLASS net/minecraft/class_3232 net/minecraft/world/gen/chunk/FlatChunkGeneratorC
 		ARG 1 structuresConfig
 	METHOD method_28916 enableLakes ()V
 	METHOD method_28917 createBiome ()Lnet/minecraft/class_1959;
+	METHOD method_29965 withLayers (Ljava/util/List;Lnet/minecraft/class_5311;)Lnet/minecraft/class_3232;
+		ARG 1 layers
+		ARG 2 structuresConfig
+	METHOD method_31926 getHeight (I)I
+		ARG 1 y
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/chunk/FlatChunkGeneratorConfig.mapping]:
```suggestion
```
This method doesn't exist (anymore)

File: mappings/net/minecraft/world/gen/feature/ReplaceBlobsFeature.mapping [hunk 6]
```diff
@@ -0,0 +1,5 @@
+CLASS net/minecraft/class_5155 net/minecraft/world/gen/feature/ReplaceBlobsFeature
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/feature/ReplaceBlobsFeature.mapping:1]: Why this rename? The registry name was not changed (same for the config class)

File: mappings/net/minecraft/world/gen/feature/RuinedPortalFeature.mapping [hunk 7]
```diff
@@ -8,10 +8,15 @@ CLASS net/minecraft/class_5183 net/minecraft/world/gen/feature/RuinedPortalFeatu
...
-	METHOD method_27211 (Ljava/util/Random;Lnet/minecraft/class_2794;Lnet/minecraft/class_5189$class_5191;ZIILnet/minecraft/class_3341;Lnet/minecraft/class_5539;)I
+	METHOD method_27211 getFloorHeight (Ljava/util/Random;Lnet/minecraft/class_2794;Lnet/minecraft/class_5189$class_5191;ZIILnet/minecraft/class_3341;Lnet/minecraft/class_5539;)I
 		ARG 0 random
 		ARG 1 chunkGenerator
 		ARG 2 verticalPlacement
+		ARG 3 airPocket
+		ARG 4 height
+		ARG 5 blockCountY
+		ARG 6 box
+		ARG 7 heightLimitView
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/feature/RuinedPortalFeature.mapping]:
```suggestion
		ARG 7 world
```

File: mappings/net/minecraft/world/gen/feature/StructureFeature.mapping [hunk 8]
```diff
@@ -36,6 +36,9 @@ CLASS net/minecraft/class_3195 net/minecraft/world/gen/feature/StructureFeature
 		ARG 2 biomeSource
 		ARG 3 worldSeed
 		ARG 5 random
+		ARG 6 pos
+		ARG 9 config
+		ARG 10 heightLimitView
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/feature/StructureFeature.mapping]:
```suggestion
		ARG 10 world
```

File: mappings/net/minecraft/world/gen/feature/ReplaceBlobsFeature.mapping (hunk 6 above)
Comment (mschae23) [reply mappings/net/minecraft/world/gen/feature/ReplaceBlobsFeature.mapping:1]: It's called "netherrack replace blobs". Minecraft uses it to replace netherrack, but you can also use a different target. So I removed that from the name to make it shorter and less confusing.

Comment (liach) [reply mappings/net/minecraft/world/gen/feature/ReplaceBlobsFeature.mapping:1]: If this `netherrack_replace_blobs` is returned by a method in this feature class, then this class should be called `NetherrackReplaceBlobsFeature` (look at advancement criterions). If this is registered like `Registry.register("netherrack_replace_blobs", new ReplaceBlobsFeature())` then `ReplaceBlobsFeature` would be better.

Comment (mschae23) [reply mappings/net/minecraft/world/gen/feature/ReplaceBlobsFeature.mapping:1]: It's registered like that in `Feature`.

**Review by jaskarth — CHANGES_REQUESTED** [review]
If it's registered as `netherrack_replace_blobs` in the feature then it should stay as NetherrackReplaceBlobsFeature

**Review by jaskarth — COMMENTED** [review]
Upon further thought, there are certain instances where renaming it is valid so it's your call here

**Comment by mschae23** [comment]
I mean, the "netherrack" doesn't matter at all in the name. It can be configured to use any block.

**Comment by liach** [comment]
Hmm, this is like naming a dimension type. You can create new instances of this feature with different registry name (like create `mymod:replace_blobs` feature with this class), yet meanwhile you can create configured versions of the `netherrack_replace_blobs` that replace blobs with different targets than netherrack. So yeah it's an either way.

Also rerequesting as to invalidate that previous change request.

**Review by jaskarth — APPROVED** [review]
Yeah, honestly it looks fine. I suppose this is why you shouldn't review PRs early morning lol

**Review by YanisBft — APPROVED** [review]
Not the end of the world

**Pull request merged**